core/src/main/java/org/gbif/pipelines/core/functions/interpretation/InterpretationException.java

```diff
@@ -0,0 +1,46 @@
+package org.gbif.pipelines.core.functions.interpretation;
+
+import org.gbif.pipelines.core.functions.interpretation.error.Issue;
+import org.gbif.pipelines.core.functions.interpretation.error.Lineage;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Interpretation Exception with issues and lineages
+ * also set interpretedObject in case you had one
+ */
+public class InterpretationException extends Exception {
+
+  private List<Issue> issues = new ArrayList<>();
```

Comment:
issues and lineages don't need be initialized here and must also be 'final'

Reply:
i added few methods (addIssue, addLineage) for convenience so not marking them as final and initializing them too. Let me know if you want this to be immutable class.